napa_app/napa_app/sdk/AlchemyAPI_NewsParams.cs

```diff
@@ -7,131 +7,89 @@ namespace AlchemyAPI
+            _commands[END_DATE] = "now";
+            _commands[COUNT] = "0";
         }
-        public void setStartDate(int amount = 0, char dateType = 'd')
+
+        public void setTaxonomy(params string[] s)
         {
-            if (amount > 0)
-            {
-                _startDate = "&start=now-" + amount + dateType;
-            }
-            else
-            {
-                _endDate = "&start=now";
-            }
+            _commands[TAXONOMY_LABEL] = String.Join("+", s);
         }
-        public void setEndDate(int amount = 0, char dateType = 'd')
+
+        public void setEntities(params Entity[] entities)
         {
-            if (amount > 0)
-            {
-                _endDate = "&end=now-" + amount + dateType;
-            }
-            else
-            {
-                _endDate = "&end=now";            
-            }
+            _commands[ENTITY_TEXT] = String.Join("+", entities.Select(e => e.Text.Replace(' ', '+')));
+            _commands[ENTITY_TYPE] = String.Join("+", entities.Select(e => e.Type));
         }
 
         public override String getParameterString()
         {
-            String retString = base.getParameterString();
-            if(_count != 0 || _count != null)
-            {
-                retString += "&count=" + _count.ToString();
-            }
-            if (_taxonomy.Count > 0)
-            {
-                retString += TAXONOMY_LABEL + string.Join("+", _taxonomy.ToArray());
-            }
-            if (_concepts.Count > 0)
-            {
-                retString += CONCEPT_TEXT + string.Join("+", _concepts.ToArray());
-            }
-            if (_keyWords.Count > 0)
-            {
-                retString += KEYWORD_TEXT + string.Join("+", _keyWords.ToArray());
-            }
-            if (_entityText.Count > 0)
-            {
-                retString += ENTITY_TEXT + string.Join("+", _entityText.ToArray());
-            }
-            if (_entityType.Count > 0)
-            {
-                retString += ENTITY_TYPE + string.Join("+", _entityType.ToArray());
-            }
-            if (_relations.Count > 0)
-            {
-                retString += RELATION + _relations;
-            }
-            if (_sentiment != null)
-            {
-                retString += SENTIMENT + "|" + _sentimentType + "score=" + _sentimentScoreEquality + _sentiment + "|";                
-            }
-            retString += _startDate + _endDate;
-            return retString;
+            return String.Format("{0}&{1}", base.getParameterString(), String.Join("&", _commands.Select(e => String.Format("{0}={1}", e.Key, e.Value))));
         }
     }
 }
+
+public enum EntityType
+{
+    Person,
+    Company
+}
+
+public class Entity
+{
+    public EntityType Type { get; set; }
+    public string Text { get; set; }
+    public Entity(EntityType type, string text)
+    {
+        Type = type;
+        Text = text;
+    }
+}
+
+public enum SentimentType
```

Comment:
Which one did you mean? Sentiment type? If so I think it's actually Negative, Neutral, Positive.
If you meant EntityType, I think those vary a bit.